### ISSUE TITLE
Force Teleport re-auth

### ISSUE THREAD

**Issue by dancmeyers** [gravitational/teleport#28368]
# What would you like Teleport to do?

I want a flag I can pass to `tsh login`, which will ignore any existing session, temporary assumed roles etc and will fetch a fresh, full length, new session with all my default roles from the Teleport cluster, including popping up a browser window to SSO login if necessary etc.

Something like `tsh login --force`.

Bonus points if I can add config to my `~/.tsh/${PROXY_HOSTNAME}.yaml` to configure a remaining session TTL where `--force` will be automatically applied even if I do not specify it.

# What problem does this solve?

When there are less than 15 minutes left of a Teleport session, attempts to `tsh aws ...` on the command line or log into an AWS web console via `Applications` on the Teleport web console are met with `Forbidden`. Attempts to re-login via `tsh login` do not work even if I have not assumed any temporary roles, the session length remains very low. To properly refresh the session time you have to `tsh logout` first, which then wipes knowledge of the proxy from the local machine, so instead of `tsh login` I have to use `tsh login --proxy very.long.and.annoying.to.type.proxy.hostname`.

I don't want to have to keep re-entering the proxy hostname whenever I need to refresh a session which I can see is very short.

I _also_ don't want to request elevated access through an access request and then find that I do not have enough time to do the necessary work because my initial session with default roles had too little time left, rather than because the max length of the elevated role is too short. I have repeatedly not noticed how little remaining TTL my session has, requested elevated access, and then by the time that gets approved my initial session (and therefore the elevated session from my access request) is < 15 minutes long and I can't do what I need with it. It would be great if I could configure `tsh` such that any time my session is < 2 hours long (arbitrary time that works for us but would probably need to be different for others) any attempt to login or request elevated rights will first issue a `tsh login --force` to get a fresh 'base' session.

# If a workaround exists, please include it.

Paying attention to remaining session TTL via `tsh status`, and `tsh logout && tsh login --proxy ${PROXY_HOSTNAME} && tsh request new ...`...

**Comment by webvictim** [comment]
This should also cover sessions start via the web UI. If someone is logged into the web UI with an active session and their certificate is approaching expiry, we should automatically refresh it so that they're not logged out of the UI in the middle of an operation.

At the very least, we should show a warning that their certificate is approaching expiry and provide a way for them to refresh it while their sessions stay active.

**Comment by GavinFrazar** [comment]
> we should automatically refresh it so that they're not logged out of the UI in the middle of an operation.

I haven't looked into this specifically, but auto-refreshing their session seems to run counter to the point of having sessions expire - I think they should have to re-authenticate. We can show a warning about session expiry though, I like that idea.

**Comment by zmb3** [comment]
IMO we shouldn't even require a `--force` flag. The user executing `tsh login` is a clear indicator that their intent is to re-login.

**Comment by ravicious** [comment]
AFAIK, `tsh login` is the only way to change the current profile, see #3089. Making every `tsh login` go through the full login procedure would make it much harder to switch between profiles when you're already logged in to at least two 2 profiles.

**Comment by GavinFrazar** [comment]
> AFAIK, `tsh login` is the only way to change the current profile, see [#3089](https://github.com/gravitational/teleport/issues/3089). Making every `tsh login` go through the full login procedure would make it much harder to switch between profiles when you're already logged in to at least two 2 profiles.

We should mimic kubectl IMO, e.g., `tsh use-profile <proxy-addr>` (ideally we'd allow people to specify custom profile names too, I don't like typing out long proxy addresses myself)

**Comment by webvictim** [comment]
`tsh switch` was also mooted in the past. Shell aliases do help with this, but having a native CLI way to switch identity with no automatic relogin is important IMO.

**Comment by carrithers** [comment]
I wonder if it would be better to take inspiration from `docker context` and offer a `tsh profile` with subcommands. For example:

`tsh profile use <profile>` 
`tsh profile show` (displays the name of the current profile) 
`tsh profile ls`
`tsh profile rm <profile>`

**Comment by ravicious** [comment]
@carrithers Yeah, I had a similar suggestion here: https://github.com/gravitational/teleport/pull/67694#discussion_r3404718002
